src/nmr_to_structure/prepare_input/prepare_nmr_rxn_input.py

```diff
@@ -1,92 +1,163 @@
+import logging
+import os
+import random
+import time
+from functools import partial
+from multiprocessing import Manager, Process
 from pathlib import Path
+from typing import List
 
 import click
 import pandas as pd
 from rxn.chemutils.tokenization import tokenize_smiles
+from rxn.utilities.logging import setup_console_and_file_logger
 from tqdm.auto import tqdm
-import random
 
-from .nmr_utils import build_cnmr_string, build_hnmr_string, save_set, split_data
+from nmr_to_structure.prepare_input.nmr_utils import (
+    DEFAULT_SEED,
+    DEFAULT_NON_MATCHING_TOKEN,
+    evaluate_molecule,
+    log_file_name_from_time,
+    make_nmr,
+    save_set,
+    split_data,
+    tanimoto_set_worker,
+)
+
+DEFAULT_TEST_SIZE = 0.1
+DEFAULT_VAL_SIZE = 0.05
+DEFAULT_MAX_SEQ_LEN = 600
+
+random.seed(DEFAULT_SEED)
 
-NON_MATCHING_TOKEN = "<no_match>"
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
 
-def make_nmr(mode: str, component: str, hnmr: dict = None, cnmr: dict = None, hnmr_mode: str = 'range', token_space: str = 'shared'):
 
-    if mode == "combined":
-        hnmr_string = build_hnmr_string(
-            smiles=component,
-            peak_dict=hnmr,
-            mode=hnmr_mode,
-            header=False,
-            token_space=token_space,
-        )[0]
+def make_nmr_tanimoto_set(
+    nmr_df: pd.DataFrame,
+    mode: str,
+    n_samples: int,
+    logger: logging.Logger,
+    cores: int = 8,
+    hnmr_mode: str = "range",
+    token_space: str = "shared",
+    non_matching=True,
+    tanimoto_bins: int = 5,
+) -> pd.DataFrame:
+    all_smiles = list(nmr_df.index)
+    random.shuffle(all_smiles)
+
+    smiles_chunks = [
+        all_smiles[
+            i
+            * len(all_smiles)
+            // cores : min((i + 1) * len(all_smiles) // cores, len(all_smiles))
+        ]
+        for i in range(cores)
+    ]
```

Comment:
For things like this one, I would typically try to use a function from another package, as it's easy to not spot a mistake. I see for instance that there is `numpy.array_split` for that purpose. Alternatively, I would put it in its own function (which has associated unit tests). See also here: https://stackoverflow.com/a/312464